### DISCUSSION TITLE
Validate Transaction (Mempool)

### DISCUSSION THREAD

**Discussion by ricore77** [ethereum-optimism/developers#615] [Application Development] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

🤔🤔🤔
Guys, does anybody know if it is possible to intercept a transaction before it enters the mempool and run a validation against a smart contract? Like a whitelist... only transactions that respect certain policies are allowed

**Comment by imtipi** [comment]
possible to hack op stack,you need to write the code yourself,theres no production ready code of transaction whitelist/filter

**Reply by ricore77** [reply]
Thank you!!!

**Reply by ricore77** [reply]
Did anybody do something similar or could point me out how to start?

**Reply by imtipi** [reply]
> Did anybody do something similar or could point me out how to start?

no as I know,cc @opfocus @joohhnnn do you guys know something about it?

**Comment by Chomtana** [comment, marked as answer]
You may need to modify the `eth_sendRawTransaction` endpoint of op-geth to check for your transaction validation requirement and return some error if it's invalid

This function should do this job
https://github.com/ethereum-optimism/op-geth/blob/8af19cf20261c0b62f98cc27da3a268f542822ee/eth/api_backend.go#L295-L320

As OP Stack is based on centralized sequencer, I think this should work

**Reply by ricore77** [reply]
Thank you 🙏🙏🙏🙏🙏
